react/AppStorePrice.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { FunctionComponent, Fragment } from 'react'
+import { useQuery } from 'react-apollo'
+import { useFormattedPrice } from 'vtex.formatted-price'
+import { AppPrice, Metric } from 'vtex.app-store-api'
+import useProduct from 'vtex.product-context/useProduct'
+import { Spinner } from 'vtex.styleguide'
+import { useIntl, defineMessages } from 'react-intl'
+import { useCssHandles } from 'vtex.css-handles'
+
+import APP_PRICE from './graphql/queries/AppPrice.graphql'
+
+const CSS_HANDLES = ['container', 'listItem', 'itemListFirst', 'itemListSecond']
+
+const messages = defineMessages({
+  metricLabel: {
+    id: 'store/app-store-price-metric-label',
+  },
+  metricLabelUpTo: {
+    id: 'store/app-store-price-metric-label-up-to',
+  },
+  metricLabelOver: {
+    id: 'store/app-store-price-metric-label-over',
+  },
+  subscriptionLabel: {
+    id: 'store/app-store-price-subscription-label',
+  },
+})
+
+const ListItem: FunctionComponent<{ paddingTop?: boolean }> = ({
+  children,
+  paddingTop,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  return (
+    <div
+      className={`${handles.listItem} flex bb b--muted-4 pb6 ${
+        paddingTop ? 'pt6' : ''
+      }`}
+    >
+      {children}
+    </div>
+  )
+}
+
+const ItemList: FunctionComponent<{
+  description: string
+  formattedValue: string
+}> = ({ description, formattedValue }) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  return (
+    <Fragment>
+      <span className={`${handles.itemListFirst} mid-gray w-60`}>
+        {description}
+      </span>
+      <span className={`${handles.itemListSecond} mid-gray w-40 f5`}>
+        {formattedValue}
+      </span>
+    </Fragment>
+  )
+}
```

Comment:
These components could use better names, such as:
```suggestion
const ListItemContainer: FunctionComponent<{ paddingTop?: boolean }> = ({
  children,
  paddingTop,
}) => {
  const handles = useCssHandles(CSS_HANDLES)
  return (
    <div
      className={`${handles.listItem} flex bb b--muted-4 pb6 ${
        paddingTop ? 'pt6' : ''
      }`}
    >
      {children}
    </div>
  )
}

const ListItemContent: FunctionComponent<{
  description: string
  formattedValue: string
}> = ({ description, formattedValue }) => {
  const handles = useCssHandles(CSS_HANDLES)
  return (
    <Fragment>
      <span className={`${handles.itemListFirst} mid-gray w-60`}>
        {description}
      </span>
      <span className={`${handles.itemListSecond} mid-gray w-40 f5`}>
        {formattedValue}
      </span>
    </Fragment>
  )
}
```